### DISCUSSION TITLE
WooCommerce product loop in Bootstrap carousel not working

### DISCUSSION THREAD

**Discussion by MartynaC** [bootscore/bootscore#226] [Q&A] [answered]
Hello,

I am using Bootscore with Woocommerce theme on my website, I am trying to add a simple product slider based on bootstrap and ACF fields.
I can not make it work, even though it works on my other websites, using different themes. In fact, I have tried many different bootstrap sliders approaches and none of it works – everything gets displayed in one column, underneath each item, I have tried these two codes for example:
https://codepen.io/mephysto/pen/ZYVKRY
https://www.jqueryscript.net/slider/multi-item-carousel-bootstrap.html

and this is the desired code, I have been successfully using on other sites:

<div id="myCarousel" class="carousel slide" data-ride="carousel">
<!– Wrapper for slides –>
<div class="carousel-inner">
<?php
$featured_posts = get_field('wyswietlane_produkty');
$is_active = true; // Only true for the first iteration
$i = 0;

if( $featured_posts ): ?>
<ul class="marka-product-listing">
<div class="row">
<?php foreach( $featured_posts as $post ):

// Setup this post for WP functions (variable must be named $post).
setup_postdata($post); ?>
<?php if ($i % 6 == 0):?>
<div class=" carousel-item item <?php if ($is_active) echo ' active'?>">
<?php endif?>
<li>

<div class="col-md-2 col-6 ">

<a href="<?php the_permalink(); ?>">
<?= get_the_post_thumbnail();?>
</a>
<?php $tags = get_terms( 'product_tag' );
if ($tags) {
foreach( $tags as $tag ) {
echo "<p><span class='text-uppercase new'>" .$tag->name. "</span></p>";
}
}
?>
<h3 class="single-marka-prod"><a href="<?php the_permalink(); ?>"><?php the_title(); ?></a></h3>
<p class="slider-price"><?= wc_get_product()->get_price_html()?></p>

</div>

</li>
<?php if (($i+1) % 6 == 0 || $i == count($featured_posts)-1):?>
</div>
<?php endif?>
<?php
$i++;
if ($is_active) $is_active = false;
endforeach; ?>
</div>
</ul>
<?php
// Reset the global post object so that the rest of the page works correctly.
wp_reset_postdata(); ?>
<?php endif; ?>
</div><!– end item—>
<a class="left carousel-control" href="#myCarousel" data-slide="prev">
<span class="glyphicon glyphicon-chevron-left"></span>
<span class="sr-only">Previous</span>
</a>
<a class="right carousel-control" href="#myCarousel" data-slide="next">
<span class="glyphicon glyphicon-chevron-right"></span>
<span class="sr-only">Next</span>
</a>
</div><!– end of item row–>

</div><!–end of carousel inner–>

Can I get some help with this one, please?
Am I missing something?

**Comment by crftwrk** [comment, marked as answer]
Yes, that‘s because Bootstrap‘s carousel component is not able to show columns by default. Getting your Codepen examples working in a WP loop with Bootstrap carousel is quiet hard.

For this reason we‘ve created the bS Swiper plugin https://bootscore.me/documentation/bs-swiper/ which is able to show WooCommerce products or any other custom post types.

Use this plugin with a simple shortcode and show your customers products in a slideshow.

BTW I converted this issue into a discussion, because this isn‘t a theme bug.

Does this help?

**Reply by MartynaC** [reply]
Thank you for your answer - the only trouble is I need to use it with ACF fields - is this achievable?

**Reply by justinkruit** [reply]
Do you want to call the same layout as on the shop page of the product (so like the single card), or do you want to make your own custom content per product item? Just asking to provide a more tailored answer 😉

Also, ACF4Life ❤️

**Reply by crftwrk** [reply]
You can use ACF fields as much as you want  in bS Swiper by adding them to the template file. Please check the documentation how to override plugin templates in your child https://bootscore.me/documentation/bs-swiper/#Overriding_templates_via_theme

**Reply by MartynaC** [reply]
I need the sliders to display content - for example products or posts or just images chosen from ACF fields, for example - I have front page, where I want to display products chosen in ACF relationship field.

**Reply by MartynaC** [reply]
I do not see how I can add that to the shortcode - I do not want sliders to be pre-made, I need to give control to my client, so he can choose his products/images from the admin. :)

**Reply by justinkruit** [reply]
You could achieve that using the plugin @crftwrk mentioned. But if you prefer to write it yourself, I highly recommend checking out SwiperJS (https://swiperjs.com/), in which you can do a loop of the relationship field.

**Reply by MartynaC** [reply]
I am not too keen on writing it myself tbh :) if there is a ready-made solution I'd be glad to use it - so if I understand it correctly, I shall overwrite the swiper file, inserting ACF fields in there and then use it as a shortcode? 

so for example if I want to create a swiper with products I shall create a template : 
/my-theme/bs-swiper-main/swiper-my-products.php

and then use it in shortcode as 

[bs-swiper-my-products  orderby="date" posts="12" ]

am i thinking correctly?

**Reply by crftwrk** [reply]
Almost, not that complicated. Plugin overrides works like WooCommerce template files, do it exactly as described in the docs.

1. Use shortcode to show products in your content `[bs-swiper-card-product category="shoes, trousers" order="DESC" orderby="date" posts="12"]`, change categor(ies) `shoes, trousers` to yours.
2. Create folder `bs-swiper-main` in your child-theme's root folder
3. Go to folder `wp-content > plugins > bs-swiper-main > templates` and copy `sc-swiper-card-product.php` straight to folder `bs-swiper-main` in your child
4. Add Advanced Custom Fields to this file

**Comment by MartynaC** [comment]
Thank you for your answers!
I manage to have it working the way I need :) 
I have to create many different sliders, so I am adding new shortcodes and creating new templates for swipers in my child theme - works great!
